osu.Game/OsuGameBase.cs

```diff
@@ -45,6 +46,8 @@ public class OsuGameBase : Framework.Game, IOnlineComponent
 
         public readonly Bindable<WorkingBeatmap> Beatmap = new Bindable<WorkingBeatmap>();
 
+        public readonly Bindable<OsuScreen> CurrentScreen = new Bindable<OsuScreen>();

```

Comment:
I really don't like exposing this, would suggest instead exposing something like `AllowBeatmapChange` on the `MusicController` and setting that in `screenChanged`.